### PR TITLE
cleanup: Remove test net support.

### PR DESCRIPTION
**Pull request by iphydf**
Mainnet now supports NGC.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2474)

### CODE REVIEW

**Comment by JFreegman** [comment]
Are we never going to need this again?

**Comment by iphydf** [comment]
> Are we never going to need this again?

Maybe we'll need something like this again, but we can add it back (in a different way) at that point. Right now it's a maintenance burden, every time having to read the correct enum when looking at packets during debugging.

**Comment by Green-Sky** [comment]
Should we allow different networks entirely? I think some dht networks use a simple passphrase.

**Comment by iphydf** [comment]
> Should we allow different networks entirely? I think some dht networks use a simple passphrase.

Yes, I think that's a good idea. I think it can be done in a backwards compatible way (accepting both with and without network id).

File: configure.ac [hunk 1]
```diff
@@ -177,10 +177,6 @@ AC_ARG_ENABLE([[test_network]],
     [use_test_network='no']
```
Comment (Green-Sky) [inline configure.ac:177]: what about this whole ARG ?

Comment (iphydf) [reply configure.ac]: Deleted. Good catch :)

**Review by Green-Sky — APPROVED** [review]

**Pull request merged**